hcl/hcl.go

```diff
@@ -131,31 +132,72 @@ type PartialEvaluator func(hclsyntax.Expression) (hclwrite.Tokens, error)
 // this API allows you to define the exact set of files (and contents) that are
 // going to be included in the final configuration.
 type TerramateParser struct {
-	dir         string
-	files       map[string][]byte // path=content
-	parsedFiles []string
-	hclparser   *hclparse.Parser
+	rootdir   string
+	dir       string
+	files     map[string][]byte // path=content
+	hclparser *hclparse.Parser
 
 	// MergedAttributes are the top-level attributes of all files.
 	MergedAttributes ast.Attributes
 
 	// MergedBlocks are the merged blocks from all files.
-	MergedBlocks map[string]*ast.MergedBlock
+	MergedBlocks ast.MergedBlocks
 
 	// Blocks are the unmerged blocks from all files.
 	Blocks ast.Blocks
+
+	// parsedFiles stores a map of all parsed files
+	parsedFiles map[string]parsedFile
+
+	// if true, calling Parse() or MinimalParse() will fail.
+	executed bool
```

Comment:
I don't like this :(

I added this protection because the parser now has a state with the merged blocks. This is far from ideal... we can improve this in the future.
Ideas are welcome!

Reply:
![](https://raw.githubusercontent.com/katcipis/memes/master/know-feeling-bro.jpg)